[3.5.0RC1] IllegalArgumentException via AuthenticatorActivity.parseLoginDataUrl

via GPlay:
```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2825)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2900)
  at android.app.ActivityThread.-wrap11 (Unknown Source)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1601)
  at android.os.Handler.dispatchMessage (Handler.java:105)
  at android.os.Looper.loop (Looper.java:251)
  at android.app.ActivityThread.main (ActivityThread.java:6589)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:240)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:767)
Caused by: java.lang.IllegalArgumentException: 
  at com.owncloud.android.authentication.AuthenticatorActivity.parseLoginDataUrl (AuthenticatorActivity.java:577)
  at com.owncloud.android.authentication.DeepLinkLoginActivity.onCreate (DeepLinkLoginActivity.java:21)
  at android.app.Activity.performCreate (Activity.java:6975)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1214)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2778)
```

piece of code causeing it:
```
for (String value : values) {
            if (value.startsWith("user" + LOGIN_URL_DATA_KEY_VALUE_SEPARATOR)) {
                loginUrlInfo.username = URLDecoder.decode(
                        value.substring(("user" + LOGIN_URL_DATA_KEY_VALUE_SEPARATOR).length()));
            } else if (value.startsWith("password" + LOGIN_URL_DATA_KEY_VALUE_SEPARATOR)) {
                loginUrlInfo.password = URLDecoder.decode(
                        value.substring(("password" + LOGIN_URL_DATA_KEY_VALUE_SEPARATOR).length()));
            } else if (value.startsWith("server" + LOGIN_URL_DATA_KEY_VALUE_SEPARATOR)) {
                loginUrlInfo.serverAddress = URLDecoder.decode(
                        value.substring(("server" + LOGIN_URL_DATA_KEY_VALUE_SEPARATOR).length()));
            } else {
                // error illegal URL element detected
                throw new IllegalArgumentException("Illegal magic login URL element detected: " + value);
            }
        }
```

To me this can only happen _if_
* Nc Login URL protocoll is correct but
* a parameter has been set that is unexpected (read: _not_ user/password/server)

In such a scenario the client should just state that (Snackbar maybe) but shouldn't report a crash on the GPlay dev console.

What do you think @tobiasKaminsky ?

--- a/src/main/java/com/owncloud/android/authentication/AuthenticatorActivity.java
+++ b/src/main/java/com/owncloud/android/authentication/AuthenticatorActivity.java
@@ -573,9 +573,6 @@ public class AuthenticatorActivity extends AccountAuthenticatorActivity
             } else if (value.startsWith("server" + LOGIN_URL_DATA_KEY_VALUE_SEPARATOR)) {
                 loginUrlInfo.serverAddress = URLDecoder.decode(
                         value.substring(("server" + LOGIN_URL_DATA_KEY_VALUE_SEPARATOR).length()));
-            } else {
-                // error illegal URL element detected
-                throw new IllegalArgumentException("Illegal magic login URL element detected: " + value);
             }
         }
 

--- a/src/test/java/com/owncloud/android/authentication/AuthenticatorDataUrlTest.java
+++ b/src/test/java/com/owncloud/android/authentication/AuthenticatorDataUrlTest.java
@@ -1,4 +1,4 @@
-/**
+/*
  *   Nextcloud Android client application
  *
  *   @author Andy Scherzinger
@@ -157,17 +157,24 @@ public class AuthenticatorDataUrlTest {
         }
     }
 
+    @Test(expected = IllegalArgumentException.class)
+    public void noDataUrlElements() {
+        AuthenticatorActivity.parseLoginDataUrl(schemeUrl, "");
+    }
+
     @Test
-    public void tooLittleDataUrlElements() {
-        for (String urlStart : urlStarts) {
-            try {
-                System.out.println(urlStart);
-                AuthenticatorActivity.parseLoginDataUrl(urlStart, urlStart);
-                Assert.fail("Illegal Argument Exception expected!");
-            } catch (IllegalArgumentException e) {
-                // all well
-                System.out.println("Expected Illegal Argument Exception caught.");
-            }
-        }
+    public void onlyOneDataUrlElements() {
+        LoginUrlInfo loginUrlInfo = AuthenticatorActivity.parseLoginDataUrl(schemeUrl, schemeUrl + userUrlPart);
+
+        Assert.assertEquals(userValue, loginUrlInfo.username);
+    }
+
+    @Test
+    public void onlyTwoDataUrlElements() {
+        LoginUrlInfo loginUrlInfo = AuthenticatorActivity.parseLoginDataUrl(schemeUrl, schemeUrl + userUrlPart +
+            plus + addressUrlPart);
+
+        Assert.assertEquals(userValue, loginUrlInfo.username);
+        Assert.assertEquals(addressValue, loginUrlInfo.serverAddress);
     }
 }

